agent is undefined

I had some python that I used claude to convert to go so I could leverage agent-api. Made a ton of progress learning how the newer go.mod works and cleaning the generated 1.16 code, but now paused on the error below. 

## error

When I run `main.go` I get the following error.

```
➜  vision git:(main) ✗ go run main.go --video input.mp4 --output frames                
# command-line-arguments
./main.go:40:53: undefined: agent.Agent
./main.go:60:53: undefined: agent.Agent
./main.go:112:18: undefined: agent.Model
➜  vision git:(main) ✗ 
```

My initial thought was ollama wasn't running, but I've checked and it is. So next is I have not set up my agent correctly. Opening this issue to add context so I can ask @jpmcb in the morning. I've just run out of gas for today.

I am also aware I've not added gsv to this, which could potentially give me  a better error 🤷🏾 

## some context on main.go

I've been floating the idea around building an agent scrapes instagram reels. The code in this repo takes a local mp4 (ideally a downloaded video and prints a description from the vision screenshots. This is a POC for work where I can get a swarm of agents to mass index saved recipes from insta. My attempt at this code is to make this generic enough that it can work with non-related workflow that needs vision agents.

--- a/main.go
+++ b/main.go
@@ -12,6 +12,7 @@ import (
 	"time"
 
 	"github.com/agent-api/core/pkg/agent"
+	"github.com/agent-api/core/types"
 	"github.com/agent-api/ollama"
 
 	"log/slog"
@@ -37,7 +38,7 @@ func extractFrames(videoPath, outputDir string, interval int) error {
 	return ffmpegCommand.Run()
 }
 
-func analyzeImage(ctx context.Context, agent *agent.Agent, imagePath string) (string, error) {
+func analyzeImage(ctx context.Context, a *agent.DefaultAgent, imagePath string) (string, error) {
 	imageData, err := os.ReadFile(imagePath)
 	if (err != nil) {
 		return "", err
@@ -49,7 +50,7 @@ func analyzeImage(ctx context.Context, agent *agent.Agent, imagePath string) (st
 		{"type": "image", "source": {"data": "%s", "media_type": "image/jpeg"}}
 	]`, imageData)
 
-	response, err := agent.Run(ctx, prompt, agent.DefaultStopCondition)
+	response, err := a.Run(ctx, prompt, agent.DefaultStopCondition)
 	if err != nil {
 		return "", err
 	}
@@ -57,7 +58,7 @@ func analyzeImage(ctx context.Context, agent *agent.Agent, imagePath string) (st
 	return response[0].Message.Content, nil
 }
 
-func processVideo(ctx context.Context, agent *agent.Agent, videoPath, outputDir string) error {
+func processVideo(ctx context.Context, a *agent.DefaultAgent, videoPath, outputDir string) error {
 	err := extractFrames(videoPath, outputDir, 5)
 	if err != nil {
 		return err
@@ -79,7 +80,7 @@ func processVideo(ctx context.Context, agent *agent.Agent, videoPath, outputDir 
 	for _, frame := range frames {
 		framePath := filepath.Join(outputDir, frame)
 		fmt.Printf("Analyzing frame: %s\n", frame)
-		analysis, err := analyzeImage(ctx, agent, framePath)
+		analysis, err := analyzeImage(ctx, a, framePath)
 		if err != nil {
 			return err
 		}
@@ -109,14 +110,16 @@ func main() {
 	provider := ollama.NewProvider(opts)
 
 	// Use the correct model
-	model := &agent.Model{Name: "llama3.2-vision:11b"}
+	model := &types.Model{
+		ID: "llama3.2-vision:11b",
+	}
 	provider.UseModel(ctx, model)
 
 	// Create agent configuration
 	agentConf := &agent.NewAgentConfig{
 		Provider:     provider,
 		Logger:       logger,
-		SystemPrompt: "You are a visual analysis assistant specialized in detailed image descriptions",
+		SystemPrompt: "You are a visual analysis assistant specialized in detailed image descriptions. If there is a person in the image describe what they are doing in step by step format.",
 	}
 
 	// Initialize agent
@@ -128,6 +131,7 @@ func main() {
 
 	err := processVideo(ctx, visionAgent, videoPath, outputDir)
 	if err != nil {
-		log.Fatal(err)
-	}
+		log.Printf("Error processing video: %v", err)
+		os.Exit(1)
+}
 }

